gapis/perfetto/android/trace.go

```diff
@@ -90,7 +88,7 @@ func setupProfileLayers(ctx context.Context, d adb.Device, driver adb.Driver, pa
 	}
 
 	// Setup render stage layer. Render stage layer should be at the bottom (end of list).
-	if hasRenderStages {
+	if hasRenderStages && d.Instance().GetConfiguration().GetPerfettoCapability().GetGpuProfiling().GetHasRenderStageProducerLayer() {
```

Comment:
Maybe just `d.Instance().GetConfiguration().GetPerfettoCapability().GetGpuProfiling().GetHasRenderStageProducerLayer()`, it implies `hasRenderStages`.

Reply:
`hasRenderStages` is the TraceConfig selection.  If the user is not trying to get render stages, then we don't need to enable the layer.